engine/source/public/core/asset.hpp

```diff
@@ -0,0 +1,70 @@
+﻿#pragma once
+#include <string>
+
+namespace lumina
+{
+    enum class AssetType
+    {
+        Texture, // Including Samplers or not can be discussed
```

Comment:
Sampler settings can be added to a texture asset, but samplers don't need their own entry in the enum unless the sampler is going to be an asset of their own

Reply:
This I was not sure about yet. Maybe we want to reuse samplers, and this would fit better

Reply:
I am okay with having the sampler as a separate asset. We would then also probably have a project setting where you can set a specific sampler as the default for when you create a new texture asset